semeio/workflows/localisation/local_script_lib.py

```diff
@@ -207,34 +210,327 @@ def activate_gen_param(
+                   can be done by specifying smooth ranges in number of
+                   grid cells in I and J direction. If this is not specified,
+                   no smoothing is done.
+                   NOTE: Smoothing is done only between active segments,
+                   and no smoothing between active segments and inactive
+                   segments or inactive grid cells.
+    """
+
+    debug_print(f"Active segments: {active_segment_list}", LogLevel.LEVEL3, log_level)
+    if scaling_factor_list is None:
+        scaling_factor_list = []
+        scaling_factor_list.append(1.0)
+        debug_print(
+            f"Scaling factors:    {scaling_factor_list}", LogLevel.LEVEL3, log_level
+        )
+
+    max_region_number_specified = 0
+    for n in active_segment_list:
+        if n > max_region_number_specified:
+            max_region_number_specified = n
+        if n < 0:
+            raise ValueError("Segment number must be non-negative integer")
+
+    region_parameter = region_param_dict[corr_name]
+    missing_code = 999
+    if missing_code in region_parameter:
+        logging.warning(
+            "Input region parameter %s has values equal to %d\n"
+            "The value is reserved as missing code internally in "
+            "the localisation script."
+        )
+    max_region_parameter = region_parameter.max()
+    if max_region_parameter < max_region_number_specified:
+        raise ValueError(
+            "Specified an active region with number "
+            f"{max_region_number_specified} which is larger \n"
+            f"than max region parameter {max_region_parameter} for "
+            f"correlation group {corr_name}."
+        )
+    # In the full nx*ny*nz region_parameter, assign a missing value to
+    # values defined as inactive in the 3D grid.
+    set_missing_code_for_inactive_grid_cells(grid, region_parameter, missing_code)
+
+    (
+        scaling_values,
+        active_localisation_region,
+        regions_in_param,
+    ) = calculate_scaling_factors_in_regions(
+        grid,
+        region_parameter,
+        missing_code,
+        active_segment_list,
+        scaling_factor_list,
+        smooth_range_list,
+    )
+    if smooth_range_list is not None:
+        scaling_values = smooth_parameter(
+            grid,
+            smooth_range_list,
+            scaling_values,
+            active_localisation_region,
+            missing_code,
         )
-    else:
-        logging.warning("Scaling method: %s is not implemented.", method)
+
+    # Assign values to row_scaling object
+    for index in range(data_size):
+        global_index = grid.global_index(active_index=index)
+        row_scaling[index] = scaling_values[global_index]
+
+    not_defined_in_region_param = []
+    for n in active_segment_list:
+        if n not in regions_in_param:
+            not_defined_in_region_param.append(n)
+    if len(not_defined_in_region_param) > 0:
+        debug_print(
```

Comment:
Should perhaps be a configuration error?

Reply:
It is not an error, but it is not  useful to define region numbers with scaling factor to be applied when it actually don't exist and has no effect.